wagtail/admin/api/actions/publish.py

```diff
@@ -0,0 +1,59 @@
+from django.core.exceptions import ValidationError as DjangoValidationError
+from django.shortcuts import get_object_or_404
+from rest_framework import fields, status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+from rest_framework.serializers import Serializer
+
+from wagtail.api.v2.utils import BadRequestError
+from wagtail.core.actions.publish_page_revision import PublishPageRevisionAction
+
+from .base import APIAction
+
+
+class PublishPageAPIActionSerializer(Serializer):
+    changed = fields.BooleanField(default=True, required=False)
+    log_action = fields.BooleanField(default=True, required=False)
+    previous_revision_id = fields.IntegerField(required=False)
+
+
+class PublishPageAPIAction(APIAction):
+    serializer = PublishPageAPIActionSerializer
+
+    def _action_from_data(self, instance, data):
+        previous_revision = None
+        user = self.request.user
+        log_action = data["log_action"]
+
+        previous_revision_id = data.get("previous_revision_id", None)
+        if previous_revision_id:
+            previous_revision = get_object_or_404(
+                instance.revisions, id=previous_revision_id
+            )
+
+        revision = instance.save_revision(
+            user=user, log_action=log_action, previous_revision=previous_revision
+        )
+
+        return PublishPageRevisionAction(
+            revision,
```

Comment:
Wonder if we should use ``instance.get_latest_revision()`` here?

Reply:
And use `save_revision` if there is no revision at all?

Reply:
Yep, thanks!